aws/resource_aws_sfn_activity.go

```diff
@@ -53,6 +55,48 @@ func resourceAwsSfnActivityCreate(d *schema.ResourceData, meta interface{}) erro
 
 	d.SetId(*activity.ActivityArn)
 
+	return resourceAwsSfnActivityUpdate(d, meta)
```

Comment:
This is trivial case now, but we prefer to always call the `Read` function after `Create` to prevent duplicate modifications on creation. When doing `Update` after `Create`, `d.IsNewResource()` conditional checks are required in the `Update` function (oft forgotten or missed during code review) to prevent the duplicate update should another attribute that is supported during `Create` and `Update` be added in the future.

When we get around to adding our own linter for the Terraform AWS provider codebase, `Update` after `Create` will be an item that is flagged.

The `Create` function should instead add the logic to call `TagResource` here, then call `Read` instead of `Update`:

```suggestion
	if v, ok := d.GetOk("tags"); ok {
		input := &sfn.TagResourceInput{
			ResourceArn: aws.String(d.Id()),
			Tags:        tagsFromMapSfn(v.(map[string]interface{})),,
		}

		log.Printf("[DEBUG] Tagging SFN Activity: %s", input)
		_, err := conn.TagResource(input)
		if err != nil {
			return fmt.Errorf("error tagging SFN Activity (%s): %s", d.Id(), input)
		}
	}

	return resourceAwsSfnActivityRead(d, meta)
```